package/MDAnalysis/analysis/dssp/dssp.py

```diff
@@ -0,0 +1,345 @@
+
+    Returns
+    -------
+    np.ndarray
+        array of '-', 'H' and 'E' symbols with secondary structure
+
+    .. versionadded:: 2.8.0
+    """
+    C3_ALPHABET = np.array(["-", "H", "E"])
+    index = np.argmax(onehot, axis=-1)
+    return C3_ALPHABET[index]
+
+
+class DSSP(AnalysisBase):
+    """Assign secondary structure using DSSP algorithm.
+
+    Analyze a selection containing a protein and assign secondary structure
+    using the Kabsch-Sander algorithm [Kabsch1983]. Only a subset of secondary
+    structure categories are implemented:
+
+    - 'H' represents a generic helix (α-helix, π-helix or 3-<sub>10</sub> helix)
+    - 'E' represents 'extended strand', participating in beta-ladder (parallel
+      or antiparallel)
+    - '-' represents unordered part
+
+    The implementation was taken from the pydssp package (v. 0.9.0)
+    https://github.com/ShintaroMinami/PyDSSP by Shintaro Minami.
+
+    Parameters
+    ----------
+    u : Universe
+        input universe
+    guess_hydrogens : bool, optional
+        whether you want to guess hydrogens positions, by default ``True``.
+        Guessing is made assuming perfect 120 degrees for all bonds that N
+        atom makes, and a N-H bond length of 1.01 A.
+        If ``guess_hydrogens`` is False, hydrogen atom positions on N atoms
+        will be parsed from the trajectory, except for the "hydrogen" atom
+        positions on PRO residues, and an N-terminal residue.
+
+    Raises
+    ------
+    ValueError
+        if ``guess_hydrogens`` is True but some non-PRO hydrogens are missing.
+
+    Examples
+    --------
+
+    For example, you can assign secondary structure for a single PDB file:
+
+    >>> from MDAnalysis.analysis.dssp import DSSP
+    >>> from MDAnalysisTests.datafiles import PDB
+    >>> import MDAnalysis as mda
+    >>> u = mda.Universe(PDB)
+    >>> run = DSSP(u).run()
+    >>> print("".join(run.results.dssp[0][:20]))
+    '--EEEEE-----HHHHHHHH'
+
+    Also, per-frame dssp assignment allows you to build average
+    secondary structure -- ``DSSP.results.dssp_ndarray`` holds
+    (n_frames, n_residues, 3) shape ndarray with one-hot encoding of
+    loop, helix and sheet, respectively:
+
+    >>> from MDAnalysis.analysis.dssp import translate
+    >>> u = mda.Universe(...)
+    >>> long_run = DSSP(u).run()
+    >>> mean_secondary_structure = translate(
+    >>>     long_run.results.dssp_ndarray.mean(axis=0)
+    >>>     )
+    >>> print(''.join(mean_secondary_structure)[:20])
+    '---HHHHHHHH---------'
+
+    .. versionadded:: 2.8.0
+    """
+
+    def __init__(self, u: Universe, guess_hydrogens: bool = True):
+        super().__init__(u.trajectory)
+        self._guess_hydrogens = guess_hydrogens
+
+        # define necessary selections
+        # O1 is for C-terminal residue
+        heavyatom_names = ("N", "CA", "C", "O O1")
+        self._heavy_atoms: dict[str, "AtomGroup"] = {
+            t: u.select_atoms(f"protein and name {t}") for t in heavyatom_names
```

Comment:
For performance reasons it may help to make an intermediate selection of `protein` first and then re-use it on line 270 and line 274
```python
protein = u.select_atoms("protein")
...
               t: protein.select_atoms(f"name {t}") for t in heavyatom_names
...
            for res in protein.residues
```
Also a bit clearer.

Reply:
yes, that works